web-ng/src/app/components/form-field-editor/form-field-editor.component.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  Component,
+  OnInit,
+  Input,
+  Output,
+  EventEmitter,
+  OnChanges,
+} from '@angular/core';
+import { FormGroup, FormBuilder } from '@angular/forms';
+import { FieldType } from '../../shared/models/form/field.model';
+
+export interface FieldTypeOptionModel {
+  icon: string;
+  label: string;
+  type: FieldType;
+}
+
+@Component({
+  selector: 'app-form-field-editor',
+  templateUrl: './form-field-editor.component.html',
+  styleUrls: ['./form-field-editor.component.css'],
+})
+export class FormFieldEditorComponent implements OnInit {
+  @Input() label?: string;
+  @Input() required?: boolean;
+  @Input() type?: string;
+  @Output() update = new EventEmitter();
+  @Output() delete = new EventEmitter();
+  fieldTypeOptions: FieldTypeOptionModel[] = [
+    {
+      icon: 'short_text',
+      label: 'Text',
+      type: FieldType.TEXT,
+    },
+    {
+      icon: 'library_add_check',
+      label: 'Select multiple',
+      type: FieldType.MULTIPLE_CHOICE,
+    },
+  ];
+
+  textFieldGroup: FormGroup;
+
+  constructor(private formBuilder: FormBuilder) {
+    this.textFieldGroup = this.formBuilder.group({
```

Comment:
What does textFieldGroup represent? Won't this component be generic, not specific to text fields?

Reply:
forgot to rename this , thanks!